tools/azure-sdk-tools/ci_tools/environment_exclusions.py

```diff
@@ -0,0 +1,407 @@
+    "azure-iot-deviceupdate",
+    "azure-digitaltwins-core",
+    "azure-eventgrid",
+    "azure-eventhub",
+    "azure-eventhub-checkpointstoreblob",
+    "azure-eventhub-checkpointstoreblob-aio",
+    "azure-eventhub-checkpointstoretable",
+    "azure-ai-formrecognizer",
+    "azure-keyvault-administration",
+    "azure-keyvault-certificates",
+    "azure-keyvault-keys",
+    "azure-keyvault-secrets",
+    "azure-developer-loadtesting",
+    "azure-maps-geolocation",
+    "azure-maps-render",
+    "azure-maps-route",
+    "azure-maps-search",
+    "azure-mixedreality-authentication",
+    "azure-ai-ml",
+    "azure-iot-modelsrepository",
+    "azure-monitor-ingestion",
+    "azure-monitor-opentelemetry-exporter",
+    "azure-monitor-query",
+    "azure-purview-administration",
+    "azure-purview-catalog",
+    "azure-purview-scanning",
+    "azure-mixedreality-remoterendering",
+    "azure-schemaregistry",
+    "azure-schemaregistry-avroencoder",
+    "azure-search-documents",
+    "azure-servicebus",
+    "azure-storage-blob",
+    "azure-storage-blob-changefeed",
+    "azure-storage-file-datalake",
+    "azure-storage-file-share",
+    "azure-storage-queue",
+    "azure-synapse-accesscontrol",
+    "azure-synapse-artifacts",
+    "azure-synapse-managedprivateendpoints",
+    "azure-synapse-monitoring",
+    "azure-synapse-spark",
+    "azure-ai-translation-document",
+    "azure-messaging-webpubsubservice",
+]
+
+
+# --------------------------------------------------------------------------------------------------------------------
+# DO NOT add packages to the below lists. They are used to omit packages that will never run type checking.
+IGNORE_FILTER = ["nspkg", "mgmt", "cognitiveservices"]
+FILTER_EXCLUSIONS = ["azure-mgmt-core"]
+IGNORE_PACKAGES = [
+    "azure-applicationinsights",
+    "azure-servicemanagement-legacy",
+    "azure",
+    "azure-storage",
+    "azure-monitor",
+    "azure-servicefabric",
+    "azure-keyvault",
+    "azure-synapse",
+    "azure-common",
+    "conda-recipe",
+    "azure-graphrbac",
+    "azure-loganalytics",
+    "azure-media-analytics-edge",
+    "azure-media-videoanalyzer-edge",
+    "azure-template",
+]
+
+
+def filter_tox_environment_string(namespace_argument: str, package_name: str) -> str:
+    """
+    Takes an incoming comma separated list of tox environments and package name. Resolves whether or not
+    each given tox environment should run, given comparison to single unified exclusion file in `environment_exclusions`.
+
+    :param namespace_argument: A namespace argument.
+    :param package_name: The name of the package. This takes the form of a comma separated list: "whl,sdist,mindepedency". "whl". "lint,pyright,sphinx".
+    """
+    if namespace_argument:
+        tox_envs = namespace_argument.strip().split(",")
+        filtered_set = []
+
+        for tox_env in tox_envs:
+            try:
+                exclusions_for_env = globals()[f"{tox_env.strip().upper()}_OPT_OUT"]
```

Comment:
@kristapratico this works great for the _explicit_ opt outs that belong to `MYPY_OPT_OUT` and the like. For `VERIFY_TYPES` opt out, that's _not_ a tox environment, so I will do a follow-up commit that adds that automatically instead of only going off of the input.